File: AtCoder/Typical90/004/004.rb
h, w = gets.chomp.split.map(&:to_i)
a = Array.new
h.times do
    a << gets.chomp.split.map(&:to_i)
end

hsum = Array.new
wsum = Array.new
a.each do |portion| hsum << portion.sum
end
a.transpose.each do |portion| wsum << portion.sum
end
for i in 0...h do
    for j in 0...w do
        print hsum[i] + wsum[j] - a[i][j]
        print j == w - 1 ? "\n" : " "
    end
end 

File: AtCoder/Typical90/004/004.cpp
#include<iostream>
#include<vector>
using namespace std;

int main(){
    int h, w;
    cin >> h >> w;
    vector<int> hsum(h, 0), wsum(w, 0);
    vector<vector<int>> a(h, vector<int>(w));
    for(int i=0; i<h; i++){
        for(int j=0; j<w; j++){
            cin >> a.at(i).at(j);
            hsum.at(i) += a.at(i).at(j);
            wsum.at(j) += a.at(i).at(j);
        }
    }

    for(int i=0; i<h; i++){
        for(int j=0; j<w; j++){
            cout << hsum.at(i) + wsum.at(j) - a.at(i).at(j);
            if(j != w-1) cout << ' ';
            else cout << endl;
        }
    }

}